src/Component/State/ProviderInterface.php

```diff
@@ -0,0 +1,27 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Component\Resource\State;
+
+use Sylius\Component\Resource\Context\Context;
+use Sylius\Component\Resource\Metadata\Operation;
+
+/**
+ * Retrieves data from a persistence layer.
+ *
+ * @experimental
```

Comment:
Is it the only class in this PR that should be experimental? Or should we remove it/add it everywhere?

Reply:
I add it on a lot of classes (on Operations for example), but indeed, maybe that'd be better to add this on every new classes on this PR.

Reply:
I added experimental tags on CallableProvider and CallableProcessor from this PR.